src/Client.php

```diff
@@ -161,29 +194,60 @@ private function post($route = '', array $body = [])
     }
 
     /**
-     * Perform cURL DELETE request to route. Used by
+     * Perform cURL GET request to route. For example used by
      * Alerts API requests: https://www.stathat.com/manual/alerts_api
      *
-     * @param string $route
-     * Ex: /x/ACCESSTOKEN/alerts/ALERTID
+     * @param string $curl_url
+     *   Ex: https://www.stathat.com/x/ACCESSTOKEN/alerts
+     * @return array or string $result
+     *   An array of alert details or the response code string in the case of a non 200 response.
      */
-    private function deleteAlertCurl($route = '')
+    private function curlGet($curl_url)
     {
-        $curlUrl = $this->alerts_url.'/'.$route;
-
         $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, $curlUrl);
-        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
-        curl_setopt($ch, CURLOPT_HEADER, true);
+        curl_setopt($ch, CURLOPT_URL, $curl_url);
         curl_setopt($ch, CURLOPT_HTTPHEADER, [
           'Accept: application/json',
           'Content-Type: application/json',
         ]);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_exec($ch);
-        $result = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        $jsonResult = curl_exec($ch);
+        $result = json_decode($jsonResult);
+        $responseCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        if ($responseCode != 200) {
+            $result = $responseCode;
+        }
+        curl_close($ch);
+
+        return $result;
+    }
+
+    /**
+     * Perform cURL DELETE request to route. For example used by
+     * Alerts API requests: https://www.stathat.com/manual/alerts_api
+     *
+     * @param string $curl_url
+     *   Ex: /x/ACCESSTOKEN/alerts/ALERTID
+     * @return string $result
+     *   Response code.
+     */
+    private function curlDelete($curl_url)
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $curl_url);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $result = curl_exec($ch);
+        $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
         curl_close($ch);
 
+        if ($http_code == 404) {
+            $result = $http_code;
+        } else {
+            $json_result = json_decode($result);
+            $result = $http_code.': '.$json_result->msg;
```

Comment:
This feels like it should maybe be a class with a `code` and `message` field? Or perhaps `200` codes return a boolean and anything else throws an exception?